crates/kitsune_p2p/kitsune_p2p/src/metrics.rs

```diff
@@ -7,6 +9,84 @@ use tokio::time::Instant;
 use crate::types::*;
 use kitsune_p2p_types::agent_info::AgentInfoSigned;
 
+use num_traits::*;
+
+/// Running average that prioritizes memory and cpu efficiency
+/// over strict accuracy.
+/// For metrics where we can't afford the memory of tracking samples
+/// for every remote we might talk to, this running average is
+/// accurate enough and uses only 5 bytes of memory.
+#[derive(Debug, Clone, Copy)]
+pub struct RunAvg(f32, u8);
```

Comment:
i don't totally understand why we "can't afford" a `u16` or ` u32` etc. here

Reply:
it's not the u32 we can't afford, it's that we can't afford to keep samples i.e. like a histogram